thenewboston_node/core/utils/atomic_write.py

```diff
@@ -0,0 +1,17 @@
+import contextlib
+import shutil
+from pathlib import Path
+
+import atomicwrites
+
+
+@contextlib.contextmanager
+def atomic_write_append(file_path, mode, **kwargs):
+    append = 'a' in mode
+    if append:
+        kwargs.setdefault('overwrite', True)
+    with atomicwrites.atomic_write(file_path, mode=mode.replace('a', 'w'), **kwargs) as fo:
+        if append and Path(file_path).exists():
+            with open(file_path, 'rb' if 'b' in mode else 'r') as cur_fo:
```

Comment:
Add TODO that this may appear to be a slow operation in which case we need to switch to operating system level file copying or redesign the need for atomic writes for append